test/metabase/upload_test.clj

```diff
@@ -1837,3 +1837,44 @@
                       (rows-for-table table)))
 
               (io/delete-file file))))))))
+
+(defn- upload-table-exists? [table]
+  ;; we don't need to worry about sql injection here
+  (-> (format "SELECT 1 FROM information_schema.tables WHERE table_name = '%s'" (:name table))
+      ((fn [sql] {:database (:db_id table), :type :native, :native {:query sql}}))
+      qp/process-query
+      :row_count
+      pos?))
+
+(defn- referencing-models []
+  (map :table_name (t2/select :information_schema.columns :column_name "table_id")))
+
+(deftest delete-upload!-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
+    (with-upload-table! [table (create-upload-table!
+                                :col->upload-type (ordered-map/ordered-map
+                                                   :_mb_row_id auto-pk-type
+                                                   :number_1 int-type
+                                                   :number_2 int-type)
+                                :rows [[1, 1]])]
+
+      (testing "The upload table and the expected application data are created"
+        (is (upload-table-exists? table))
+        (is (seq (t2/select :model/Table :id (:id table))))
+        (is (seq (t2/select :model/Field :table_id (:id table))))
+        ;; there are some other miscellaneous children
+        ;; FIXME for some reason I get some stuff populated here locally but not in CI - not critical to debug
```

Comment:
Ah, it turns out the only other table was `v_fields` - perhaps that view is not created, or is updated more quickly, in CI.